python/cudf/cudf/core/dataframe.py

```diff
@@ -1123,7 +1123,15 @@ def __setitem__(self, arg, value):
                     for col_name in self._data:
                         self._data[col_name][mask] = value
             else:
-                if isinstance(value, DataFrame):
+                if isinstance(value, (cupy.ndarray, np.ndarray)):
+                    _setitem_with_dataframe(
+                        input_df=self,
+                        replace_df=cudf.DataFrame(value),
```

Comment:
Any chance we can make use of factory method `Dataframe._from_data` here instead of public ctor?

Reply:
I did check this but realized we have the logic that handles `ndarray`'s in `DataFrame` constructor and will need this type of inputs to go through that code-paths.